File: 面试题22-链表中倒数第k个节点.py
class Solution:
    def getKthFromEnd(self, head: ListNode, k: int) -> ListNode:
        cur_node = head
        for i in range(k):
            cur_node = cur_node.next
        
        while cur_node:
            head = head.next
            cur_node = cur_node.next
        
        return head

File: 面试题22-链表中倒数第k个节点.cpp
class Solution {
public:
    ListNode* getKthFromEnd(ListNode* head, int k) {
        ListNode* cur_node = head;
        for(int i = 0; i < k; i++){
            cur_node = cur_node->next;
        }

        while(cur_node){
            head = head->next;
            cur_node = cur_node->next;
        }

        return head;
    }
};
